### PR TITLE
Logging

### LINKED ISSUE #41: Client: creating dataset does not give error message when non-priveleged user tries creating a dataset
I attempted to create a dataset when authenticated as my user (calcuttj) and it returned a blank line. I could not see the dataset in the list of datasets (as expected). 

After authenticating as dunepro, I tried running the create dataset command again and only then it created the dataset (as expected).

To be clear: My user should not be able to create the dataset, but an error/warning should be shown

### PR DESCRIPTION
**Pull request by alisonpeisker**
Enabled more metacat client error messages to be logged:
- Log messages will now appear in the log files for each metacat instance.
- Made the ordering of the error code and message consistent across all the methods in data_handler.py.
- Also fixed #41. Added an error message for create_dataset as mentioned in the issue, and for create_namespace and add_child_dataset as well.

Included some minor improvements to the testing:
- Added a test to create a namespace for the username, and to test that it fails if the namespace already exists.
- Increased the number of files in the test dataset to help with the metacat dataset add files test, so if the user running the tests doesn't have any files in metacat yet it this test should still work.
- Removed a test that looked like an exact duplicate.

### CODE REVIEW

File: tests/test_commandline.py [hunk 1]
```diff
@@ -97,6 +97,13 @@ def test_metacat_auth_export(auth):
 #    with os.popen("metacat auth import", "r") as fin:
 #        data = fin.read()
 
+# need to first create a namespace for the username
+# also test that the second time it's created fails
+def test_initial_namespace_create(auth):
+    os.popen(f'metacat namespace create {os.environ["USER"]}')
```
Comment (marcmengel) [inline tests/test_commandline.py:103]: Can we make this one  os.system() instead of  os.popen()?  Otherwise we're leaking a file descriptor...

**Review by marcmengel — CHANGES_REQUESTED** [review]
Just one small change requested.

File: tests/test_commandline.py (hunk 1 above)
Comment (alisonpeisker) [reply tests/test_commandline.py]: Yes, I changed it. Thanks for catching that.

**Review by marcmengel — APPROVED** [review]
Looks great, Thanks!

**Pull request merged**